modules/fedimint-wallet-client/src/pegin_monitor.rs

```diff
@@ -0,0 +1,488 @@
+                if is_running_in_test_env() {
+                    // In tests, we basically mine all blocks right away
+                    Some(Duration::from_millis(1))
+                } else {
+                    Some(Duration::from_secs(60 * num_blocks_needed))
+                }
+            }
+            // Once anything has been claimed, there's no reason to claim again automatically,
+            // and it's undesirable due to privacy reasons.
+            // Users can possibly update the underlying record via other means to force a check on
+            // demand.
+            CheckOutcome::Claimed { .. } | CheckOutcome::AlreadyClaimed => None,
+        }
+    }
+
+    /// Desired retry delay for a bunch of outcomes.
+    ///
+    /// This time is intended to be persisted in the database.
+    ///
+    /// None means "no need to check anymore".
+    fn retry_delay_vec(outcomes: &[CheckOutcome], creation_time: SystemTime) -> Option<Duration> {
+        // If the address was allocated, but nothing was ever received or even detected
+        // on it yet, check again in time proportional to the age of the
+        // address.
+        if is_running_in_test_env() {
+            // When testing we usually send deposits right away, so check more aggressively.
+            return Some(Duration::from_millis(100));
+        } else if outcomes.is_empty() {
+            let now = time::now();
+            let age = now.duration_since(creation_time).unwrap_or_default();
+            return Some(age / 2);
+        }
+
+        // The delays is the minimum retry delay.
+        let mut min = None;
+
+        for outcome in outcomes {
+            min = match (min, outcome.retry_delay()) {
+                (None, time) => time,
+                (Some(min), None) => Some(min),
+                (Some(min), Some(time)) => Some(cmp::min(min, time)),
+            };
+        }
+
+        min
+    }
+
+    fn get_claimed_now_outpoints(outcomes: &[CheckOutcome]) -> Vec<bitcoin::OutPoint> {
+        let mut res = vec![];
+        for outcome in outcomes {
+            if let CheckOutcome::Claimed { outpoint } = outcome {
+                res.push(*outpoint);
+            }
+        }
+
+        res
+    }
+}
+
+/// Query via btc rpc for a history of an address derived with `tweak_idx` and
+/// claim any peg-ins that are ready.
+///
+/// Return a list of [`CheckOutcome`]s for each matching output.
+#[instrument(skip_all, fields(tweak_idx))]
+async fn check_idx_pegins(
+    data: &WalletClientModuleData,
+    tweak_idx: TweakIdx,
+    btc_rpc: &DynBitcoindRpc,
+    module_rpc: &DynModuleApi,
+    db: &Database,
+    client_ctx: &ClientContext<WalletClientModule>,
+) -> Result<Vec<CheckOutcome>, anyhow::Error> {
+    let current_consensus_block_count = module_rpc.fetch_consensus_block_count().await?;
+    let (script, address, tweak_key, operation_id) = data.derive_peg_in_script(tweak_idx);
+    btc_rpc.watch_script_history(&script).await?;
```

Comment:
Probably fine to keep as-is since the client likely won't use bitcoind anytime soon, but we call `watch_script_history` when we allocate a deposit address, so I don't believe this is necessary and I'm also curious about the original intent behind this comment.

https://github.com/fedimint/fedimint/blob/449f030e78455b69e7422ac0fb71f46d4e4614c8/fedimint-bitcoind/src/lib.rs#L172-L173

Reply:
> , so I don't believe this is necessary and I'm also curious about the original intent behind this comment.

What if the underlying bitcoind changed or something. I don't think there's a reason to not call it - I doubt the performance matters, and if the bitcoind already imported it, it will just immediately responde with "OK".

Maybe we should get rid of the whole call altogether.